### ISSUE TITLE
TypeError: Path must be a string. Received undefined.

### ISSUE THREAD

**Issue by seanfisher** [kennu/serverless-plugin-autoprune#1]
Right after installing this plugin on a serverless project version 0.5.1 I receive the following error upon running any Serverless cli command:

```
/usr/local/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:49
    fn = function () { throw arg; };
                       ^

TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.join (path.js:1253:7)
    at module.exports (/Users/sean/prog/node_modules/serverless-plugin-autoprune/index.js:12:32)
    at Serverless._loadPlugins (/usr/local/lib/node_modules/serverless/lib/Serverless.js:286:25)
    at Serverless.loadProjectPlugins (/usr/local/lib/node_modules/serverless/lib/Serverless.js:240:10)
    at /usr/local/lib/node_modules/serverless/lib/Serverless.js:77:41
From previous event:
    at Serverless.init (/usr/local/lib/node_modules/serverless/lib/Serverless.js:73:25)
    at Object.<anonymous> (/usr/local/lib/node_modules/serverless/bin/serverless:17:12)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:142:18)
    at node.js:939:3
```

Is this because of the big Serverless update to version 0.5 or is this something else? Thanks!

**Comment by kennu** [comment]
Thanks for reporting. The problem seems to be caused by Serverless 0.5 compatibility and I've published a new version of serverless-plugin-autoprune (0.5.1) that should fix it. Please let me know if it works!

**Comment by seanfisher** [comment]
The `TypeError` is gone now, but now I'm getting:

```
$ sls function deploy myFunction
Serverless: Deploying functions in "dev" to the following regions: us-west-2
Serverless: ------------------------
Serverless: Successfully deployed functions in "dev" to the following regions:
Serverless: us-west-2 ------------------------
Serverless:   myFunction (project-myFunction): arn:aws:lambda:us-west-2:28438*******:function:project-myFunction:dev
{ [ResourceNotFoundException: Function not found: arn:aws:lambda:us-west-2:57312*******:function:project-myFunction]
  cause:
   { [ResourceNotFoundException: Function not found: arn:aws:lambda:us-west-2:57312*******:function:project-myFunction]
     message: 'Function not found: arn:aws:lambda:us-west-2:57312*******:function:project-myFunction',
     code: 'ResourceNotFoundException',
     time: Sat Mar 26 2016 18:26:51 GMT-0500 (CDT),
     requestId: '38059d7a-f3aa-11e5-b612-91151117cc9c',
     statusCode: 404,
     retryable: false,
     retryDelay: 92.3834775108844 },
  isOperational: true,
  code: 'ResourceNotFoundException',
  time: Sat Mar 26 2016 18:26:51 GMT-0500 (CDT),
  requestId: '38059d7a-f3aa-11e5-b612-91151117cc9c',
  statusCode: 404,
  retryable: false,
  retryDelay: 92.3834775108844 }
```

It looks as though the plugin is using a different account number in the lambda ARN than the one serverless is using to deploy it. I have several AWS credentials in `~/.aws/credentials`, maybe it's grabbing a different one than Serverless is configured to use?

**Comment by kennu** [comment]
I think I got it, there's a new way to call AWS APIs in Serverless 0.5. Published serverless-plugin-autoprune 0.5.2 which is using the new way.

**Comment by seanfisher** [comment]
Sweet! Thanks for the fast response (and great plugin). It seems to be working. I'll open another issue if something else comes up.
